_benchmark/workloads/vectorsearch.md

```diff
@@ -0,0 +1,572 @@
+| `query_k`                                 | The number of neighbors to return for the search. Only one of `query_k`, `query_max_distance`, and, `query_min_score` can be provided.          |
+| `query_max_distance `                     | The maximum distance to be returned for the vector search. Only one of `query_k`, `query_max_distance`, and, `query_min_score` can be provided. |
+| `query_min_score`                         | The minimum score to be returned for the vector search. Only one of `query_k`, `query_max_distance`, and, `query_min_score` can be provided.    |
+| `query_data_set_format`                   | The format of the vector dataset used for queries. Only one of `query_k`, `query_max_distance`, and, `query_min_score` can be provided.        |
+| `query_data_set_path`                     | The path to the vector dataset used for queries.                                                                                               |
+| `query_count`                             | The number of queries for the search operation.                                                                                                 |
+| `query_body`                              | The JSON properties that will be merged with search body                                                                                        |
+| `search_clients`                          | The number of clients to use for running queries.                                                                                               |
+| `repetitions`                             | THe number of repetitions until the dataset is exhausted. Default is `1`.                                                                      |
+| `target_throughput`                       | The target throughput for each query operation in requests per second. Default is `10`.                                                         |
+| `time_period`                             | The period of time dedicated for the run the benchmark test in seconds. Default is `900`.                                                       |
+
+
+
+## Test procedures
+
+The workload supports the following test procedures.
+
+### No train test
+
+The no train test procedure tests vector search indices that require no training. You can define the underlying configuration of the vector search algorithm (such as specific engine or space type) as method definitions. 
+
+### No train test (Index only)
+
+The no train test (Index only) procedure is used to index vector search indexes that require no training. It's particularly useful when you want to benchmark only the indexing operation.
+
+### No train test (AOSS)
+
+The No train test (AOSS) procedure is similar to the No Train Test procedure but is specifically targeted for Amazon OpenSearch Serverless vector search collection. This procedure doesn't include operations like **refresh** and **warmup** since they aren't supported by Vector Search Collection.
+
```

Comment:
Global: Are the names of the procedures supposed to be initially capitalized? Sometimes they are and sometimes they aren't. We would also need to define AOSS, which Amazon does not use as an abbreviation for Amazon OpenSearch Serverless, FYI.

Reply:
I still AOSS crop up in Dev drafts from time to time since its the prefix for Serverless (https://docs.aws.amazon.com/service-authorization/latest/reference/list_amazonopensearchserverless.html). Noted. 

As far as the names of the procedures, there isn't a rule ATM. Likely less confusing if we lowercase them.

Reply:
I added suggestions fixing the AOSS note. Let me know what you want to do with the casing of procedures.